schemas/stsci.edu/asdf/0.1.0/core/column.yaml

```diff
@@ -0,0 +1,49 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/column"
+tag: "tag:stsci.edu:asdf/0.1.0/core/column"
+
+title: >
+  A column in a table.
+
+description: |
+  Each column contains a name and an array of data, and an optional description
+  and unit.
+
+type: object
+properties:
+  name:
+    description: |
+      The name of the column.  Each name in a
+      [table](http://stsci.edu/schemas/asdf/0.1.0/core/table) must be
+      unique.
+    type: string
+    pattern: "[A-Za-z_][A-Za-z0-9_]*"
+
+  data:
+    description: |
+      The array data for the column.
+    allOf:
+      - $ref: ndarray
+
+  description:
+    description: |
+      An optional description of the column.
+    type: string
+    default: ''
+
+  unit:
```

Comment:
I like this idea too.  FITS tables have something similar as well (it's never been supported in pyfits, though I would like to).  Definitely agree it should be optional, but I also have no reservations about keeping representation hints in the metadata (ASDF schemas include such information as well).